src/main/java/uk/gov/companieshouse/api/accounts/service/impl/SmallFullServiceImpl.java

```diff
@@ -1,34 +1,135 @@
+            String companyAccountId, String requestId)
+            throws DataException {
+
+        String selfLink = createSelfLink(transaction, companyAccountId);
+        initLinks(smallFull, selfLink);
+        smallFull.setEtag(GenerateEtagUtil.generateEtag());
+        smallFull.setKind(Kind.SMALL_FULL_ACCOUNT.getValue());
+        SmallFullEntity baseEntity = smallFullTransformer.transform(smallFull);
+
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("transaction_id", transaction.getId());
+        debugMap.put("company_accounts_id", companyAccountId);
+
+        try {
+            baseEntity.setId(generateID(companyAccountId));
+            smallFullRepository.insert(baseEntity);
+            companyAccountService.addLink(companyAccountId, LinkType.SMALL_FULL, selfLink);
+        } catch (DuplicateKeyException dke) {
+            LOGGER.errorContext(requestId, dke, debugMap);
+            return new ResponseObject<>(ResponseStatus.DUPLICATE_KEY_ERROR, null);
+        } catch (MongoException me) {
+            DataException dataException = new DataException(
+                    "Failed to insert " + ResourceName.SMALL_FULL.getName(), me);
+            LOGGER.errorContext(requestId, dataException, debugMap);
+            throw dataException;
+        }
+
+        return new ResponseObject<>(ResponseStatus.CREATED, smallFull);
     }
 
     @Override
-    public void addKind(SmallFull rest) {
-        rest.setKind(Kind.SMALL_FULL_ACCOUNT.getValue());
+    public ResponseObject<SmallFull> findById(String id) {
+        SmallFullEntity smallFullEntity = smallFullRepository.findById(id).orElse(null);
+        if (smallFullEntity == null) {
+            return new ResponseObject<>(ResponseStatus.NOT_FOUND);
+        }
+        SmallFull smallFull = smallFullTransformer.transform(smallFullEntity);
+        return new ResponseObject<>(ResponseStatus.FOUND, smallFull);
     }
 
     @Override
-    public String getResourceName() {
-        return "small-full";
+    public String generateID(String value) {
+        String unencryptedId = value + "-" + ResourceName.SMALL_FULL.getName();
+        byte[] id = messageDigest.digest(
+                unencryptedId.getBytes(StandardCharsets.UTF_8));
+        return Base64.getUrlEncoder().encodeToString(id);
     }
 
+    @Override
+    public void initLinks(SmallFull smallFull, String link) {
+        Map<String, String> map = new HashMap<>();
+        map.put(LinkType.SELF.getLink(), link);
+        smallFull.setLinks(map);
+    }
+
+    @Override
+    public String createSelfLink(Transaction transaction, String companyAccountId) {
+        return transaction.getLinks().get(LinkType.SELF.getLink()) + "/company-account/"
+                + companyAccountId + "/" + ResourceName.SMALL_FULL.getName();
+    }
+
+    @Override
+    public void addLink(String id, LinkType linkType, String link) {
+        SmallFullEntity smallFullEntity = smallFullRepository.findById(id)
```

Comment:
as above - should be throwing DataException from findById and passing up the chain here